content/300-guides/025-migrate/500-data-migration.mdx

```diff
@@ -0,0 +1,260 @@
+  status    Status   @default(Unknown)
+}
+
+enum Status {
+  Unknown
+  Draft
+  InProgress
+  InReview
+  Published
+}
+```
+
+Re-run `prisma migrate dev` to generate and apply a migration to the database.
+
+## Create and run the data migration file
+
+### Create a data migration file
+
+Inside the generated migration folder from the previous step, create a file called `data-migration.ts` file. This file will contain a data migration which will be implemented using Prisma Client.
+
+Add the following code to migrate the data from the `published` field to the `status` field in the file you just created:
+
+Update your `package.json` file to include the data migration file:
+
+```ts file=prisma/migrations/20230417131956_add-status-column/data-migration.ts
+import { PrismaClient } from '@prisma/client'
+
+const prisma = new PrismaClient()
+
+async function main() {
+  await prisma.$transaction(async (tx) => {
+    const posts = await tx.post.findMany()
+    for (const post of posts) {
+      await tx.post.update({
+        where: { id: post.id },
+        data: {
+          status: post.published ? 'Published' : 'Unknown',
+        }
+      })
+    }
+  })
+}
+
+main()
+  .catch(async (e) => {
+    console.error(e)
+    process.exit(1)
+  }).finally(async () => await prisma.$disconnect()
+  )
+```
+
+The data migration is wrapped in a transaction to ensure that the query is rolled back, allowing you to iterate on your data migration file
+
+
+Next steps:
+
+1. Push your changes to a remote origin and create a new pull request.
+1. Once you’re happy with the changes, merge the changes to your `main` branch.
+
+To apply the changes to your production database, add `prisma migrate deploy` as part of your deployment/ build step in CI
+
+
+### Run the data migration
+
+Update the `package.json` file with the script to execute the data-migration file. Be sure to update the `20230417131956_add-status-column` with the name of your migration file.
+
+```json file=package.json
+"scripts": {
+    "dev": "ts-node ./script.ts",
+    "data-migration:add-status-column": "ts-node ./prisma/migrations/20230417131956_add-status-column/data-migration.ts"
+  },
+```
+
+Next steps:
+
+1. Push your changes to a remote origin and create a new pull request.
+1. Once you’re happy with the changes, merge the changes to your “main” branch.
+
+To apply the changes to your production database, add `prisma migrate deploy` as part of your deployment/ build step in CI.
+
+### Run the data migration
+
+Update the `DATABASE_URL` environment variable with your production database's URL. Run the data migration script:
+
+```bash
+npm run data-migration:add-status-column
+```
+
+## Contract your schema by dropping the old column
+
+Checkout to a separate branch on your development machine:
+
+```bash
+git checkout -b drop-published-column
+```
+
+Delete the `published` field from your schema and generate a new migration:
+
+```prisma highlight=12;delete
```

Comment:
another issue with the line highlight: 

<img width="697" alt="image" src="https://github.com/prisma/docs/assets/4058327/1ac56f8b-e98c-4d5d-bef2-e6193b4e144d">